File: 2015/python/d02.py
lines = open('d02.in').read().split()

area = 0
ribbon = 0
for line in lines:
    x, y, z = sorted(map(int, line.split('x')))
    area += 2 * (x * y + x * z + y * z) + x * y
    ribbon += x + x + y + y + x * y * z

print('P1:', area)
print('P2:', ribbon)

File: 2020/julia/d02.jl
function readinput(filename)
    map(group -> filter(!isempty, split(group, " ")), readlines(filename))
end

lines = readinput("d02.in")

P1, P2 = 0, 0

for line in lines
    nums = parse.(Int, split(line[1], '-'))
    char = line[2][1]
    pass = line[3]
    x = count(x -> x == char, pass)
    if nums[1] <= x <= nums[2]
        global P1 += 1
    end
    if xor(pass[nums[1]] == char, pass[nums[2]] == char)
        global P2 += 1
    end
end

println("P1: ", P1)
println("P2: ", P2)
